core/base/inc/TString.h

```diff
@@ -44,21 +45,27 @@ TString operator+(const TString &s1, const TString &s2);
 TString operator+(const TString &s,  const char *cs);
 TString operator+(const char *cs, const TString &s);
 TString operator+(const TString &s, char c);
-TString operator+(const TString &s, Long_t i);
-TString operator+(const TString &s, ULong_t i);
-TString operator+(const TString &s, Long64_t i);
-TString operator+(const TString &s, ULong64_t i);
 TString operator+(char c, const TString &s);
-TString operator+(Long_t i, const TString &s);
-TString operator+(ULong_t i, const TString &s);
-TString operator+(Long64_t i, const TString &s);
-TString operator+(ULong64_t i, const TString &s);
 Bool_t  operator==(const TString &s1, const TString &s2);
 Bool_t  operator==(const TString &s1, const char *s2);
 Bool_t  operator==(const TSubString &s1, const TSubString &s2);
 Bool_t  operator==(const TSubString &s1, const TString &s2);
 Bool_t  operator==(const TSubString &s1, const char *s2);
 
+template<class T>
+struct is_signed_numeral : std::integral_constant<bool,
+   std::is_integral<T>::value && std::is_signed<T>::value && !std::is_same<T,Char_t>::value
+> {};
+
+template<class T>
+struct is_unsigned_numeral : std::integral_constant<bool,
+   std::is_integral<T>::value && !std::is_signed<T>::value && !std::is_same<T,Char_t>::value
+> {};
+
+template<class T>
+struct is_float_numeral : std::integral_constant<bool,
```

Comment:
What does that add over `std::is_floating_point<T>`?

Reply:
I wanted to have a consistent naming scheme. I have simplified it with a `using` alias:

```cpp
template<class T>
using is_float_numeral = std::is_floating_point<T>;
```